### PR TITLE
added post functions(create,edit & delete)

### PR DESCRIPTION
**Pull request by anasm11**
skip config,css and backend files.

### CODE REVIEW

File: src/contexts/PostContext.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import axios from "axios"
+import {useEffect,useReducer,createContext} from "react"
+
+const PostContext=createContext()
+
+const postReducer=(state,{type,payload})=>{
+    switch(type){
+        case ('UPDATE'):{
```
Comment (kaustubh-jsr) [inline src/contexts/PostContext.js:8]: You can add default fail-safe case to this switch statement here.

**Review by kaustubh-jsr — APPROVED** [review]
The code is cleanly written, could not find any issues, just added a suggestion. You can go ahead and merge.

File: src/components/CreatePost.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    return (
+        <div className='p-3 bg-gray-100 rounded-lg border border-gray-200 shadow-md sm:p-6 lg:p-8 dark:bg-gray-800 dark:border-gray-700'>
+
+            <textarea className='textarea p-2 w-full bg-white rounded border border-black' onChange={(e) => setPostContent(e.target.value)}></textarea>
+            <button className='button' onClick={async () => {
+                setIsCreatePostVis(false)
+                const res = await createPostApiCall(postContent);
+                dispatchPost({ type: 'UPDATE', payload: res.data.posts })
+                console.log(res.data)
```
Comment (yaswanthmyneni) [inline src/components/CreatePost.jsx]: changes: make a separate function for the onClick and also remove console.log here

File: src/components/EditModal.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+        <div className='p-3 bg-gray-100 rounded-lg border border-gray-200 shadow-md sm:p-6 lg:p-8 dark:bg-gray-800 dark:border-gray-700'>
+
+            <textarea className='textarea p-2 w-full bg-white rounded border border-black'
+                onChange={(e) => setPostToBeEdited((post) => { return { ...post, content: e.target.value } })} value={postToBeEdited.content}></textarea>
+            <button className='button' onClick={async () => {
+                setShowEditModal(null)
+                const res = await updatePostApiCall(post._id, postToBeEdited.content);
+                dispatchPost({ type: 'UPDATE', payload: res.data.posts })
+                console.log(res.data)
```
Comment (yaswanthmyneni) [inline src/components/EditModal.jsx]: same here and everywhere

File: src/utils/postApiCalls.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            authorization: localStorage.getItem('token')
+        },
+        data: { postData: { content: postContent } }
+    
+        
+    })
+    return res
+}
+    catch(e){
+        console.log(e,'error')
```
Comment (yaswanthmyneni) [inline src/utils/postApiCalls.js]: suggestion: use console.error for displaying errors

**Review by yaswanthmyneni — APPROVED** [review]
you can merge after changes

**Pull request merged**